plugin-infra/go-plugin-access/src/main/java/com/thoughtworks/go/plugin/access/configrepo/ConfigRepoExtensionContract.java

```diff
@@ -29,6 +29,6 @@
 public interface ConfigRepoExtensionContract {
 
     CRParseResult parseDirectory(String pluginId, final String destinationFolder, final Collection<CRConfigurationProperty> configurations);
-    String pipelineExport(String pluginId, final CRPipeline pipelineConfig);
+    ExportedConfig pipelineExport(String pluginId, final CRPipeline pipelineConfig);
```

Comment:
We now return an `ExportedConfig` object, rather than a plain string, to encapsulate both content and metadata together (i.e., content type and filename) in order to support automatic download.